### PR TITLE
Implement BlockchainConnection.getTx

### LINKED ISSUE #1008: Use LightTransaction
With https://github.com/iov-one/iov-core/pull/975 merged, we should do the following `BlockchainConnection` changes:

- [x] Create
`getTransaction: (id: TransactionId) => Promise<ConfirmedTransaction | FailedTransaction>;`.
This can later be used when clicking on transaction details from a transactions list. With the result of that, it should be possible to verify the signature locally.
- [x] Change interface to
`searchTx(query: TransactionQuery): Promise<ReadonlyArray<LightTransaction | FailedTransaction>>`
- [x] Change interface to
`listenTx(query: TransactionQuery): Stream<LightTransaction | FailedTransaction>`
- [x] Change interface to
`liveTx(query: TransactionQuery): Stream<LightTransaction | FailedTransaction>`

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #1008

Probably the tests verifying signatures are unnecessary but I added them as a sanity check. Happy to remove them if they’re just clutter.

### CODE REVIEW

File: packages/iov-bns/src/bnsconnection.spec.ts [hunk 1]
```diff
@@ -938,6 +939,101 @@ describe("BnsConnection", () => {
...
+      {
+        const nonExistentId = "abcd" as TransactionId;
+        await connection
+          .getTx(nonExistentId)
+          .then(fail.bind(null, "should not resolve"), error =>
+            expect(error).toMatch(/transaction does not exist/i),
+          );
+      }
+
+      // by existing ID (from lisk/init.sh)
```
Comment (webmaster128) [inline packages/iov-bns/src/bnsconnection.spec.ts]: lisk/init.sh in BNS tests?

File: packages/iov-ethereum/src/ethereumconnection.spec.ts [hunk 2]
```diff
@@ -691,17 +691,122 @@ describe("EthereumConnection", () => {
...
+      {
+        const nonExistentId = "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef" as TransactionId;
+        await connection
+          .getTx(nonExistentId)
+          .then(fail.bind(null, "should not resolve"), error =>
+            expect(error).toMatch(/transaction does not exist/i),
+          );
+      }
+
+      // by existing ID (from lisk/init.sh)
```
Comment (webmaster128) [inline packages/iov-ethereum/src/ethereumconnection.spec.ts]: lisk/init.sh?

**Comment by webmaster128** [comment]
> What about moving them in the implementation of getTx and document getTx to be signature verifying? Seems to be a strong feature.

hmm, maybe a bit out of scope and when the signature is already verified, there is no real reason to return it.

File: packages/iov-bns/src/bnsconnection.spec.ts (hunk 1 above)
Comment (willclarktech) [reply packages/iov-bns/src/bnsconnection.spec.ts]: Whoops, leftover from copy/pasting the test structure.

**Comment by willclarktech** [comment]
Nice idea, but I think it makes more sense to have separate functions for retrieval and verification. Do we have an issue already to provide a nice wrapper for this kind of verification?

**Comment by webmaster128** [comment]
> Do we have an issue already to provide a nice wrapper for this kind of verification?

Nope, signature verification has never been on the TODO

**Comment by willclarktech** [comment]
Then I guess we wait until after MVP?

**Pull request merged**